README.md

```diff
@@ -1,271 +1,215 @@
+This section will contain information about all the ML driven optimizations. Here is a brief about each optimization, and a simple onnx command which we can use to get one output (i.e give it an input .c/.cpp/.ll and get the optimized binary) .
 
-`python gen-odg.py <Path_to_opt> -Oz`
+> [!TIP] 
+> if you'd like to see the LLVM IR that is resulted from these optimization , you can pass the appropriate flags to generate the .ll files
 
-The graph and sub-sequences can be generated for other LLVM optimization levels. The required optimization flag needs to be provided as an argument when calling the above script.
+### Reinforcement Learning assisted Loop Distribution for Locality and Vectorization
 
-## Experiments
-Install and activate the conda environment
+We propose a Reinforcement Learning (RL) approach for loop distribution, optimizing for both vectorization and locality. Using SCC Dependence Graphs (SDGs), our RL model learns loop distribution order through topological walks. The reward is based on instruction cost and cache misses. We introduce a strategy to expand the training set by generating new loops. This method aims to enhance loop parallelization and improve overall code performance.
 
-`conda env create -f rllib_env.yml`
+This is described in the paper [here](https://ieeexplore.ieee.org/abstract/document/10026979) .
+Please see [here](https://compilers.cse.iith.ac.in/publications/rl_loop_distribution/) for more details.
 
-`conda activate rllib_env`
+> Reinforcement Learning assisted Loop Distribution for Locality and Vectorization, Shalini Jain, S. VenkataKeerthy, Rohit Aggarwal, Tharun Kumar Dangeti, Dibyendu Das, Ramakrishna Upadrasta
 
-Use `-mcpu=cortex-a72` for AArch64 architecture when calling `clang` or `opt` in (RLLib-PhaseOrder/Environment.py)[RLLib-PhaseOrder/Environment.py]
+#### Try it out !!!
 
-### Training
+> We assueme you have already done the setup and built the project.
 
-Add path to directory containing LLVM IR files to be used for training in [RLLib-PhaseOrder/Environment.py](RLLib-PhaseOrder/Environment.py)
+```bash
+# ONNX command for inference:
+# this script will generate the optimized llfile
+./build/bin/opt -S \
+	-custom_loop_distribution \
+	-cld-use-onnx \
+	-ml-config-path= # path to your ml config  \
+	<file name> 
+```
+to learn more head to the Pass specific readme [here].
 
-`python experiment.py --llvm_dir <Path to llvm build directory> --ir2vec_dir <Path to directory with IR2Vec binary and seed embedding>`
+### RL4Real
 
-### Inference
+`RL4ReAl` is a retargetable Reinforcement Learning (RL) approach for solving the REgister ALlocation (REAL) problem on diverse architectures.
 
-Add paths to `llvm_dir`, `ir2vec_dir` and saved RLLib model to run-inference.sh
+This is described in the paper [here](https://dl.acm.org/doi/abs/10.1145/3578360.3580273) ([arXiv](https://arxiv.org/abs/2204.02013)).
+Please see [here](https://compilers.cse.iith.ac.in/publications/rl4real/) for more details.
 
```

Comment:
Please add reference to the paper like what you have in distribution and poset-rl